nucleus/deploy/client.py

```diff
@@ -263,24 +274,31 @@ def list_model_endpoints(self) -> List[AsyncModelEndpoint]:
             A list of ModelEndpoint objects
         """
         resp = self.connection.get(ENDPOINT_PATH)
-        return [
-            AsyncModelEndpoint(endpoint_id=endpoint_id, client=self)
-            for endpoint_id in resp["endpoints"]
+        async_endpoints: List[Union[AsyncModelEndpoint, SyncModelEndpoint]] = [
+            AsyncModelEndpoint(endpoint_id=endpoint["name"], client=self)
+            for endpoint in resp["endpoints"]
+            if endpoint["endpoint_type"] == "async"
+        ]
+        sync_endpoints: List[Union[AsyncModelEndpoint, SyncModelEndpoint]] = [
+            SyncModelEndpoint(endpoint_id=endpoint["name"], client=self)
+            for endpoint in resp["endpoints"]
+            if endpoint["endpoint_type"] == "sync"
         ]
+        return async_endpoints + sync_endpoints
```

Comment:
Don't the set of async and sync endpoints comprise the full set of possible endpoints? If so, can we just return `resp["endpoints"]`?

Reply:
It is the full set of endpoints, but the object returned is a list of either `SyncModelEndpoints` or `AsyncModelEndpoints`, so we have to construct them separately (or do something like `modelEndpointClass = SyncModelEndpoint if <x> else Async`)